### PR TITLE
Add .NET 8 Targets

### PR DESCRIPTION
**Pull request by rhysparry**
# Background

With the EOL of .NET 6 coming in November, this PR sets up Tentacle to build versions targeting .NET 8.

# Results

- Adds a .NET8 version of the Tentacle (and other tooling)

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/main/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Octopus.Tentacle.Tests.Integration/Support/RuntimeDetection.cs
Comment (rhysparry) [inline source/Octopus.Tentacle.Tests.Integration/Support/RuntimeDetection.cs:41]: nit: Calling `GetCurrentRuntime()` twice unless running on net6.0

File: source/Octopus.Tentacle.Tests.Integration/Octopus.Tentacle.Tests.Integration.csproj
Comment (michaelongso) [inline source/Octopus.Tentacle.Tests.Integration/Octopus.Tentacle.Tests.Integration.csproj:23]: Was there a failure that led us to add this property?

File: source/Octopus.Tentacle.Tests.Integration/ScriptServiceV2IntegrationTest.cs
Comment (michaelongso) [inline source/Octopus.Tentacle.Tests.Integration/ScriptServiceV2IntegrationTest.cs:173]: ℹ️ This was done to reduce flakiness on Win2012 integration tests.
- It might stay/remove depending on the decision whether to run win2012 build

File: build/Build.Tests.cs
Comment (michaelongso) [inline build/Build.Tests.cs:102]: The removal here is validated against the public doc on the [supported Linux distro](https://octopus.com/docs/infrastructure/deployment-targets/linux#supported-distributions).

File: build/Build.cs
Comment (michaelongso) [inline build/Build.cs:297]: ℹ️ We made a modification to this file before its creation. Now, we made sure it is created so we can capture the modification.

File: build/ModifiableFileWithRestoreContentsOnDispose.cs
Comment (rhysparry) [inline build/ModifiableFileWithRestoreContentsOnDispose.cs:30]: ℹ️ Make sure we only restore the file contents once.

File: build/test-scripts/integration/common/run-integration-tests-on-vm-agents.sh
Comment (rhysparry) [inline build/test-scripts/integration/common/run-integration-tests-on-vm-agents.sh:16]: ℹ️ If `$TARGET_FRAMEWORK` is not set, it will default to `net8.0`

File: build/test-scripts/integration/common/setup-vm-agent.sh
Comment (rhysparry) [inline build/test-scripts/integration/common/setup-vm-agent.sh:19]: We probably want to look into this to only do a one line change as we only need the 8.0 sdk and not the 6.0 runtime.

I.e.

```
install_dotnet --sdk 8.0
```

File: build/test-scripts/integration/run-kubernetes-tests-on-vm-agents.sh
Comment (rhysparry) [inline build/test-scripts/integration/run-kubernetes-tests-on-vm-agents.sh:16]: ❓ This might be worth changing to use an environment variable (like `run-integration-tests-on-vm-agents.sh`)

File: source/Octopus.Manager.Tentacle.Tests/AssemblyInfo.cs
Comment (rhysparry) [inline source/Octopus.Manager.Tentacle.Tests/AssemblyInfo.cs:3]: Things complains about using Windows only APIs for this Windows only feature (Tentacle Manager).

File: source/Octopus.Manager.Tentacle.Tests/Octopus.Manager.Tentacle.Tests.csproj
Comment (rhysparry) [inline source/Octopus.Manager.Tentacle.Tests/Octopus.Manager.Tentacle.Tests.csproj:33]: ℹ️ This was copied as it was clearer than adding a more complex condition

File: source/Octopus.Manager.Tentacle/Octopus.Manager.Tentacle.csproj
Comment (rhysparry) [inline source/Octopus.Manager.Tentacle/Octopus.Manager.Tentacle.csproj:21]: This is now in a conditional group below

File: source/Octopus.Manager.Tentacle/Octopus.Manager.Tentacle.csproj
Comment (rhysparry) [inline source/Octopus.Manager.Tentacle/Octopus.Manager.Tentacle.csproj:149]: We're making sure these old system package references are no longer used with net8.0

File: source/Octopus.Tentacle.Client.Tests/Octopus.Tentacle.Client.Tests.csproj
Comment (rhysparry) [inline source/Octopus.Tentacle.Client.Tests/Octopus.Tentacle.Client.Tests.csproj:19]: Remove the TODO. Only in test project

File: source/Octopus.Tentacle.Kubernetes.Tests.Integration/Octopus.Tentacle.Kubernetes.Tests.Integration.csproj
Comment (rhysparry) [inline source/Octopus.Tentacle.Kubernetes.Tests.Integration/Octopus.Tentacle.Kubernetes.Tests.Integration.csproj:15]: Remove the todo. This is in tests only

File: source/Octopus.Tentacle.Tests.Integration.Common/Octopus.Tentacle.Tests.Integration.Common.csproj
Comment (rhysparry) [inline source/Octopus.Tentacle.Tests.Integration.Common/Octopus.Tentacle.Tests.Integration.Common.csproj:15]: Remove the todo, only used by tests

File: source/Octopus.Tentacle.Tests.Integration/Octopus.Tentacle.Tests.Integration.csproj
Comment (rhysparry) [reply source/Octopus.Tentacle.Tests.Integration/Octopus.Tentacle.Tests.Integration.csproj:23]: Yes. Remove the property and see. Remove the todo as it only impacts tests

File: source/Octopus.Tentacle.Tests.Integration/ScriptServiceV2IntegrationTest.cs
Comment (rhysparry) [reply source/Octopus.Tentacle.Tests.Integration/ScriptServiceV2IntegrationTest.cs:173]: Revert shi

File: source/Octopus.Tentacle.Tests.Integration/Support/IntegrationTestTimeout.cs
Comment (rhysparry) [inline source/Octopus.Tentacle.Tests.Integration/Support/IntegrationTestTimeout.cs:1]: Revert this

File: source/Octopus.Tentacle.Tests.Integration/Support/TentacleFetchers/NugetTentacleFetcher.cs
Comment (rhysparry) [inline source/Octopus.Tentacle.Tests.Integration/Support/TentacleFetchers/NugetTentacleFetcher.cs:81]: With everything on .NET 8 we might not need this any more.

File: source/Octopus.Tentacle.Tests.Integration/TentacleCommandLineTests.cs
Comment (rhysparry) [inline source/Octopus.Tentacle.Tests.Integration/TentacleCommandLineTests.cs:1]: Revert this

File: source/Octopus.Tentacle.Tests.Integration/Util/SilentProcessRunnerFixture.cs
Comment (rhysparry) [inline source/Octopus.Tentacle.Tests.Integration/Util/SilentProcessRunnerFixture.cs:1]: Revert this

Comment (rhysparry) [reply source/Octopus.Tentacle.Tests.Integration/Util/SilentProcessRunnerFixture.cs:1]: Or don't and reference the Octopus Server PR where the same change was made

File: source/Octopus.Tentacle.Tests/Octopus.Tentacle.Tests.csproj
Comment (rhysparry) [inline source/Octopus.Tentacle.Tests/Octopus.Tentacle.Tests.csproj:43]: Remove this line. Tests only.

File: source/Octopus.Tentacle.Upgrader/Octopus.Tentacle.Upgrader.csproj
Comment (rhysparry) [inline source/Octopus.Tentacle.Upgrader/Octopus.Tentacle.Upgrader.csproj:38]: Might need to revert this one to be windows only.

File: source/Octopus.Tentacle/Octopus.Tentacle.csproj
Comment (rhysparry) [inline source/Octopus.Tentacle/Octopus.Tentacle.csproj:63]: Check if this has been updated in main

File: source/Octopus.Tentacle/Security/AesEncryption.cs
Comment (rhysparry) [inline source/Octopus.Tentacle/Security/AesEncryption.cs:78]: NET8 requires the encryption algorithm is explicit as the other overload has been marked as obsolete.

Internally, it used the SHA1 algorithm.

File: source/Octopus.Tentacle/Util/SilentProcessRunner.cs
Comment (rhysparry) [inline source/Octopus.Tentacle/Util/SilentProcessRunner.cs:297]: Check if this should be `NET8_0_OR_GREATER` instead

File: source/Octopus.Tentacle.Tests.Integration/Octopus.Tentacle.Tests.Integration.csproj
Comment (michaelongso) [reply source/Octopus.Tentacle.Tests.Integration/Octopus.Tentacle.Tests.Integration.csproj:23]: I've removed the TODO and replaced with,

File: source/Octopus.Tentacle.Tests.Integration/Util/SilentProcessRunnerFixture.cs
Comment (michaelongso) [reply source/Octopus.Tentacle.Tests.Integration/Util/SilentProcessRunnerFixture.cs:1]: Reverted. (Will observe any build issue)

File: source/Octopus.Tentacle.Tests.Integration/Support/TentacleFetchers/NugetTentacleFetcher.cs
Comment (michaelongso) [reply source/Octopus.Tentacle.Tests.Integration/Support/TentacleFetchers/NugetTentacleFetcher.cs]: This one is sus. 2 attempts to remove the NET6 reference resulted in a massive test failure.
-https://github.com/OctopusDeploy/OctopusTentacle/pull/991/commits/42fd80ad0b46dd83310b85007ce87d342bfc864a#diff-902a65266fdc53157ba08edabaff0a144b82f643004614fd6aabd3cb72f247e7
- https://github.com/OctopusDeploy/OctopusTentacle/pull/991/commits/a0c7f459d563ef54145cca5a7601dd8121cd91c4#diff-902a65266fdc53157ba08edabaff0a144b82f643004614fd6aabd3cb72f247e7

CC: @acodrington

Comment (michaelongso) [reply source/Octopus.Tentacle.Tests.Integration/Support/TentacleFetchers/NugetTentacleFetcher.cs]: At this point, I'll revert the change first to get a green build.

**Comment by michaelongso** [comment]
We reviewed this work as a mob.
cc: @rhysparry, @acodrington

**Review by MissedTheMark — APPROVED** [review]
MD code owner review: approved! ✅ 
Note that this isn't a review of the functionality, the PR still requires a full review from the originating team.

File: source/Octopus.Tentacle.Tests.Integration/Support/TentacleFetchers/NugetTentacleFetcher.cs
Comment (michaelongso) [reply source/Octopus.Tentacle.Tests.Integration/Support/TentacleFetchers/NugetTentacleFetcher.cs:81]: This is likely to be the reason, https://docs.google.com/document/d/1bVnkDoctYpf-GdkoAPzGTc5Oyj_O2NlwqRl43Gv3MwM/edit#bookmark=id.976uxf7b549q
- I didn't do any further investigation though.

**Review by michaelongso — APPROVED** [review]

**Pull request merged**
